negy-gateway/src/main.rs

```diff
@@ -66,9 +69,12 @@ async fn fetch_nodes_unselected(
         .filter(|n| {
             if let Some(min_version) = &min_version {
                 Version::parse(&n.version).unwrap() >= Version::parse(min_version).unwrap()
-            } else {
-                true
-            }
+            } else {true}
+        })
+        .filter(|n| {
+            if let Some(block_network) = &block_network {
+                block_network[0..block_network.len()].contains(block_network)
```

Comment:
I don't think this gonna work correctly since the `block_network` is a `String`. Did you confirm the behavior? You should split the arg by `,` (comma) first.